[jit] serialize autograd ops into its own namespace

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26761 [jit] serialize autograd ops into its own namespace**

Summary:
This PR serialize autograd ops into its own namespace by turning the
serialization op name into `torch.autograd.op`, this is to keep the
original code namespace rather than turning all to the global namespace,
this will be more properly handled in the future when we handle the module
namespace. This change also preserve BC until we have namespace handling

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17645438](https://our.internmc.facebook.com/intern/diff/D17645438)

wanchaol committed:
[jit] serialize autograd ops into its own namespace

Summary:
This PR serialize autograd ops into its own namespace by turning the
serialization op name into `torch.autograd.op`, this is to keep the
original code namespace rather than turning all to the global namespace,
this will be more properly handled in the future when we handle the module
namespace. This change also preserve BC until we have namespace handling

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

zdevito commented on torch/csrc/jit/passes/python_print.cpp:
```diff
@@ -934,11 +934,26 @@ struct PythonPrintPass {
   }
 
   void printOpName(TaggedStringStream& stmt, Symbol kind) {
+    const static std::unordered_set<Symbol> autograd_ns_symbols = {
```
This feels overly specific to autograd, it can just be a map from symbol to an override for how it is serialized:
```
{aten::backward, "torch.autograd.backward"},
```

wanchaol committed:
address comments on "[jit] serialize autograd ops into its own namespace"


Summary:
This PR serialize autograd ops into its own namespace by turning the
serialization op name into `torch.autograd.op`, this is to keep the
original code namespace rather than turning all to the global namespace,
this will be more properly handled in the future when we handle the module
namespace. This change also preserve BC until we have namespace handling

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

wanchaol committed:
rebase on "[jit] serialize autograd ops into its own namespace"


Summary:
This PR serialize autograd ops into its own namespace by turning the
serialization op name into `torch.autograd.op`, this is to keep the
original code namespace rather than turning all to the global namespace,
this will be more properly handled in the future when we handle the module
namespace. This change also preserve BC until we have namespace handling

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
